unitVaryTime/timeVaryingTesting.md

```diff
@@ -0,0 +1,115 @@
+	// stepTime = moment.duration(1, 'hour');
+	meterUnit = 10;
+	graphicUnit = 1;
+	slope = 1;
+	stepSlope = 0.1;
+	// Move endTime back one stepTime so can check vs. startTime.
+	endTime.subtract(stepTime);
+	// Loop over all times wanted.
+	while (startTime.isSameOrBefore(endTime)) {
+		// First one really should be -infinity and last infinity but not doing now.
+		// insert into cik values(10, 1, 2, 0, '-infinity', '2021-06-02 00:00:00');
+		// Output a cik to insert into DB. Note it also steps the start time to get the end time so ready for next iteration.
+		console.log(`insert into cik values(${meterUnit}, ${graphicUnit}, ${slope.toFixed(2)}, 0, '${startTime.format("YYYY-MM-DD HH:mm:ss")}', '${startTime.add(stepTime).format("YYYY-MM-DD HH:mm:ss")}');`);
+		// Next slope.
+		slope += stepSlope;
+	}
+}
+
+module.exports = {
+	createHourly
+}
+```
+4. Before creating the data, make sure to adjust your `` startTime `` and `` endTime `` to the desired time length you want to test. Also make sure to change the `` stepTime `` to either `` 'day `` or `` 'hour' ``, depending on what you want to test time wise. Check the `` meterUnit `` number. For testing, we used `` Electric_Utility `` which was meter number 10.
+
+To check what number your  `` Electric_Utility `` is, navigate to your OED Database Shell and first enter `` psql -U oed ``. 
+
+Then enter the following: `` select * from units; `` It will bring up a table of all of the meters and find your `` Electric Utility `` meter number.
+
+
+5. Now, navigate to your OED Web Shell and type in the following npm command:
+`` npm run createCikTest ``. This will generate the test data. Make sure to copy and paste all of your test data into a outside document, this will make it easier when you insert the data into your OED Database Shell.
+
+
+
+## Inserting Data and Running Tests
+Now that you have gotten your test data, we will go over how to insert it to run the queries to test said data.
+
+1. Navigate to your OED Database Shell. If you haven't done so, type `` psql -U oed ``.
+
+2. For timing, simply type `` \timing ``. This will show the amount of time (ms) it takes to run the queries.
+
+3. Make sure your `` CIK `` table is empty before inserting the test data. To do this simply use: 
+
+``` sql
+ select * from cik where source_id = 10 and destination_id = 1; 
+```
+
+This will display your `` CIK `` table. You can also run this to simple get a count of what is in the `` CIK `` table:
+
+``` sql 
+select count(*) from cik where source_id = 10 and destination_id = 1; 
+```
+
+If there is items in your `` CIK `` simply run to remove anything from the `` CIK ``:
+
+``` sql 
+delete from cik where source_id = 10 and destination_id = 1; 
+```
+
+Note: In the queries provided above, we have a `` source_id = 10 `` and `` destination_id = 1 ``. The `` source_id `` being used for this is `` Electric_Utility `` and `` destination_id `` being used is `` Electric Utility kWh ``. 
```

Comment:
I think CIK has the source as you describe but the destination is a graphic unit so it is kWh. Do you agree?